packages/core/scripts/local/EncodeParams.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @notice The purpose of this script is to encode ExpiringMultiParty parameters.
+ *
+ * Spinning up mainnet fork: ganache-cli --fork https://mainnet.infura.io/v3/5f56f0a4c8844c96a430fbd3d7993e39 --unlock 0x2bAaA41d155ad8a4126184950B31F50A1513cE25 --unlock 0x7a3a1c2de64f20eb5e916f40d11b01c441b2a8dc --port 9545
+ * Example: yarn truffle exec ./packages/core/scripts/local/EncodeParams.js --cversion latest --network mainnet-fork
+ */
+const { toWei } = web3.utils;
+const { ZERO_ADDRESS } = require("@uma/common");
+const { getAbi, getTruffleContract } = require("../../index");
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["cversion"]
+});
+const abiVersion = argv.cversion || "latest"; // Default to most recent mainnet deployment, 1.2.2.
+
+// Deployed contract ABI's and addresses we need to fetch.
+const ExpiringMultiPartyCreator = getTruffleContract("ExpiringMultiPartyCreator", web3, abiVersion);
+const ExpiringMultiParty = getTruffleContract("ExpiringMultiParty", web3, abiVersion);
+const Finder = getTruffleContract("Finder", web3, abiVersion);
+const TokenFactory = getTruffleContract("TokenFactory", web3, abiVersion);
+
+/** ***************************************************
+ * Main Script
+ /*****************************************************/
+const encodeParams = async callback => {
```

Comment:
why is this a callback and not just return the result or throw the error naturally?

Reply:
When run with truffle exec, it has to be written this way. If we want to run it _outside_ of truffle exec (similar to how we run the bots) we could avoid this weird syntax. If we were to do that, not sure if we should put this script in a separate dir or something.